mantidproject/mantid
# Remove legacy code

**Description of work.**

This removed the Legacy base classes created during the migration of the tabs in the Indirect Data Analysis GUI. It also updates all the tests which used to test these old Legacy base classes to now test the new base classes.



**To test:**
  * Code review
  * Have a brief check that the Indirect->DataAnalysis Gui still functions correctly/


Fixes #28057 


*This does not require release notes* because this is a developer facing change

---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
